libarchive/test/test_write_format_zip_compression_bzip2.c

```diff
@@ -343,8 +343,12 @@ static void verify_bzip2_contents(const char *buff, size_t used)
 	archive_read_free(zip_archive);
 }
 
+#endif /* HAVE_BZLIB_H */
 DEFINE_TEST(test_write_format_zip_compression_bzip2)
 {
+#ifndef HAVE_BZLIB_H
```

Comment:
Another approach would be to have `archive_write_zip_set_compression_bzip2()` return `ARCHIVE_FAILED` if it cannot set the compression because bzlib is not available.  Libarchive generally does that throughout (instead of having the function not be present) for exactly this reason.

Reply:
Ah.  I see that's already implemented:  I would have this test check the result of `archive_write_set_compression_bzip2()` and skip the test if that returns `ARCHIVE_FAILED` instead of relying on `HAVE_*` symbols that tests should not be using in any case.

Reply:
Precisely - moving away from `HAVE_` checks would be good both in terms of code clarity as well as consistent test coverage. Aka we get the same number of tests, regardless of how libarchive is built - just the test status changes.

This has the extra benefit that people can grab the tests from master and run them against the older libarchive on their system. Where the failures will highlight how desperately they need to update ;-)